cli/deno_dir.rs

```diff
@@ -318,105 +325,276 @@ impl DenoDir {
           get_cache_filename(self.deps_http.as_path(), &j).as_ref(),
         )
       }
-      // TODO(kevinkassimo): change this to support other protocols than http
+      // TODO(kevinkassimo): change this to support other protocols than http.
       _ => unimplemented!(),
     }
 
     debug!("module_name: {}, filename: {}", module_name, filename);
     Ok((module_name, filename))
   }
-}
 
-impl SourceMapGetter for DenoDir {
-  fn get_source_map(&self, script_name: &str) -> Option<Vec<u8>> {
-    match self.fetch_module_meta_data(script_name, ".", true) {
-      Err(_e) => None,
-      Ok(out) => match out.maybe_source_map {
-        None => None,
-        Some(source_map) => Some(source_map),
-      },
+  /// This fetches source code, locally or remotely.
+  /// module_name is the URL specifying the module.
+  /// filename is the local path to the module (if remote, it is in the cache
+  /// folder, and potentially does not exist yet)
+  ///
+  /// It *does not* fill the compiled JS nor source map portions of
+  /// ModuleMetaData. This is the only difference between this function and
+  /// fetch_module_meta_data_async(). TODO(ry) change return type to reflect this
+  /// fact.
+  ///
+  /// If this is a remote module, and it has not yet been cached, the resulting
+  /// download will be written to "filename". This happens no matter the value of
+  /// use_cache.
+  fn get_source_code_async(
+    self: &Self,
```

Comment:
I'd prefer if we could keep these functions out of DenoDir and rather provide explicit parameters for whatever it needs. (During a recent refactoring in this file I discovered these methods to be rather tangled up without a particularly good reason.)
"filename" should already give the location of the cache. You should be able to use that for resolving the "mime file" ?